qt/widgets/instrumentview/src/XIntegrationControl.cpp

```diff
@@ -126,9 +126,92 @@ bool XIntegrationScrollBar::eventFilter(QObject *object, QEvent *e) {
       }
     }
     return true;
+  } else if (e->type() == QEvent::KeyPress) {
+    QKeyEvent *keyEv = static_cast<QKeyEvent *>(e);
+    int step = 1;
+    int sliderx = m_slider->x();
+    int slidery = m_slider->y();
+    int sliderWidth = m_slider->width();
+    int sliderHeight = m_slider->height();
+    int totalWidth = this->width();
+
+    switch (keyEv->key()) {
+    case Qt::Key_Left:
+      if (sliderx >= step) {
+        m_slider->move(sliderx - step, slidery);
+      } else {
+        m_slider->move(0, slidery);
+      }
+      m_changed = true;
+      updateMinMax();
+      break;
+
+    case Qt::Key_Right:
+      if (sliderx + sliderWidth + step < totalWidth) {
+        m_slider->move(sliderx + step, slidery);
+      } else {
+        m_slider->move(totalWidth - sliderWidth, slidery);
+      }
+      m_changed = true;
+      updateMinMax();
+      break;
+
+    case Qt::Key_Up:
+      // widen the range
+
+      // expand to the left, depending on the space on this side
+      if (sliderx > step) {
+        m_slider->resize(sliderWidth + step, sliderHeight);
+        m_slider->move(sliderx - step, slidery);
+      } else {
+        m_slider->resize(sliderWidth + sliderx, sliderHeight);
+        m_slider->move(0, slidery);
+      }
+
+      // then expand to the right, depending on the space on this side
+      if (sliderx + sliderWidth + step <= totalWidth) {
+        m_slider->resize(m_slider->width() + step, sliderHeight);
+      } else {
+        m_slider->resize(totalWidth - m_slider->x(), sliderHeight);
+      }
+      m_changed = true;
+      updateMinMax();
+      break;
+
+    case Qt::Key_Down:
+      // shrink the range
+
+      // only change the range if it is not already minimal
+      if (sliderWidth > m_resizeMargin) {
+
+        // shrink depending on how far the range is from being minimal
+        if (sliderWidth - 2 * step >= m_resizeMargin) {
+          m_slider->move(sliderx + step, slidery);
+          m_slider->resize(sliderWidth - 2 * step, sliderHeight);
+        } else {
+          m_slider->move(sliderx + (sliderWidth - m_resizeMargin) / 2, slidery);
+          m_slider->resize(m_resizeMargin, sliderHeight);
+        }
+      }
+      m_changed = true;
+      updateMinMax();
+      break;
+
+    default:
+      break;
+    }
+    return true;
+  } else if (e->type() == QEvent::KeyRelease) {
+    QKeyEvent *keyEv = static_cast<QKeyEvent *>(e);
+    if ((keyEv->key() == Qt::Key_Left || keyEv->key() == Qt::Key_Right ||
+         keyEv->key() == Qt::Key_Up || keyEv->key() == Qt::Key_Down) &&
+        !keyEv->isAutoRepeat()) {
+      emit changed(m_minimum, m_maximum);
+    }
+    return true;
   }
   return false;
-}
+} // namespace MantidWidgets
```

Comment:
This  comment seems to be a leftover, can you remove it?

Reply:
Done